src/gtk/textctrl.cpp

```diff
@@ -124,6 +124,41 @@ static void wxGtkTextApplyTagsFromAttr(GtkWidget *text,
         }
     }
 
+    if( attr.HasFontUnderlinedWithEffect() )
+    {
+        GtkTextTag *tag1;
+        PangoUnderline pg_style;
+        if( !(attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_NONE || attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_SOLID ) )
+        {
+            if( attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_NONE )
+                pg_style = PANGO_UNDERLINE_NONE;
+            if( attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_SOLID )
+                pg_style = PANGO_UNDERLINE_SINGLE;
+            if( attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_DOUBLE )
+                pg_style = PANGO_UNDERLINE_DOUBLE;
+            if( attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_WAVE )
+                pg_style = PANGO_UNDERLINE_ERROR;
+        }
+
+        g_snprintf(buf, sizeof(buf), "WXFONTUNDERLINEWITHEFFECT");
```

Comment:
How exactly does this work when we don't use this tag anywhere? What is it for?

Reply:
@vadz,
We are using the "buf" variable in the very next line in order to check if the tag had been created already.